### PR TITLE
fix: update jpi version 0.43.0, gradle version 6.0.1, buildPlugin() command, and table-to-div migration

### PR DESCRIPTION
**Pull request by yasokada-lc**
fix: update gradle version 6.0.1
fix: update jpi version 0.43.0
fix: update buildPlugin command

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by yasokada-lc** [comment]
In some use cases, multiple scm is needed instead of pipline.
Appropriate maintenance is desired.

**Comment by nneul** [comment]
@yasokada-lc - does your MR happen to fix the complete breakage of this plugin in 2.277.x jenkins instances? Just filed JENKINS-65644 - with 2.277.x - looks like you cannot add a new SCM at all - the web UI just does nothing. No javascript error.

**Comment by nneul** [comment]
Just tried building it locally - unfortunately, looks like it does not help the issue with 2.277.x

**Comment by yasokada-lc** [comment]
I running test browser chrome or fire fox
Jenkins 2.263.3

2.277.x ???

oh no.
screen01-plugin-management
screen02-buildjob-sample

**Comment by yasokada-lc** [comment]
Jenkins 2.263.3 running
but Jenkins 2.277.4 is loops ininity setTimeout(async)

Jenkins 2.263.3
screen03-jenkins-2 263 3

Jenkins 2.277.4
screen04-jenkins-2 277 4

**Comment by yasokada-lc** [comment]
I updated the hetero-list-scm file and it works.

screen05-jenkins-2 277 4-ok
screen06-jenkins-2 277 4-version-footerk

**Comment by nneul** [comment]
That fixes it for my 2.277.4 instance with latest build!

**Comment by nneul** [comment]
Now question is what it would take to get someone to actually merge this......

**Comment by nneul** [comment]
I've requested it in https://issues.jenkins.io/browse/JENKINS-65644 - hopefully someone will look at it.

**Comment by nneul** [comment]
I did a build myself to validate on my server. You can try it if you want:

https://drive.google.com/file/d/1EapOnTjS8qzrZz6EpTMaTQ31oI6M_7n5/view?usp=sharing

**Comment by nneul** [comment]
Quickest way to build if you want to be safe:  

```
git clone git@github.com:jenkinsci/multiple-scms-plugin.git
cd multiple-scms-plugin
git checkout feature/update-jpi
docker run --rm -u root -v "$PWD":/home/gradle/project -w /home/gradle/project gradle gradle build
```

result will be in build/libs

**Comment by timja** [comment]
> Quickest way to build if you want to be safe:
> 
> ```
> git clone git@github.com:yasokada-lc/multiple-scms-plugin.git
> cd multiple-scms-plugin
> git checkout feature/update-jpi
> docker run --rm -u root -v "$PWD":/home/gradle/project -w /home/gradle/project gradle gradle build
> ```
> 
> result will be in build/libs

Or use the CI build 😄 https://ci.jenkins.io/job/Plugins/job/multiple-scms-plugin/job/PR-26/3/artifact/build/libs/multiple-scms.hpi

File: build.gradle [hunk 1]
```diff
@@ -12,12 +12,13 @@ description = 'Allows multiple SCM plugins to be used in a job.'
...
 
+// Follow https://github.com/jenkinsci/gradle-jpi-plugin#configuration
 jenkinsPlugin {
     // version of Jenkins core this plugin depends on, must be 1.420 or later
-    coreVersion = '1.580.1'
+    jenkinsVersion = '2.264'
     shortName = 'multiple-scms'
     displayName = 'Jenkins Multiple SCMs plugin'
-    url = 'https://wiki.jenkins-ci.org/display/JENKINS/Multiple+SCMs+Plugin'
+    url = 'https://plugins.jenkins.io/multiple-scms/'
```
Comment (timja) [inline build.gradle:21]: this should be the github repo, it's used for documentation
```suggestion
    url = 'https://github.com/jenkins/multiple-scms-plugin'
```

Comment (timja) [reply build.gradle:21]: the wiki content should be exported and added to the repo first before changing this URL https://jenkins-wiki-exporter.jenkins.io/

**Review by timja — APPROVED** [review]

File: src/test/java/org/jenkinsci/plugins/multiplescms/BuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+public class BuilderTest {
+
+    @Rule
+    public JenkinsRule jenkins = new JenkinsRule();
+
+    final String name = "Bobby";
+
+    @Test
+    public void testConfigRoundtrip() throws Exception {
```
Comment (timja) [inline src/test/java/org/jenkinsci/plugins/multiplescms/BuilderTest.java:21]: this doesn't appear to do anything?

File: build.gradle (hunk 1 above)
Comment (jimklimov) [reply build.gradle]: Thanks, I'll address the docs in another PR then :)

File: src/test/java/org/jenkinsci/plugins/multiplescms/BuilderTest.java (hunk 2 above)
Comment (jimklimov) [reply src/test/java/org/jenkinsci/plugins/multiplescms/BuilderTest.java:21]: I believe this satisfies a minimal formality level to get the plugin built - that there *is* some test to run and report to inspect.

Comment (jimklimov) [reply src/test/java/org/jenkinsci/plugins/multiplescms/BuilderTest.java:21]: That said, I would welcome actual tests appearing eventually... though don't hold my breath for them with the plugin's outlook gloom as people migrate from legacy (freestyle etc.) jobs where this plugin really helps, into pipelines that can just use many checkout steps.

**Review by jimklimov — APPROVED** [review]
Thanks! Modulo the minimal required version of Jenkins core, looked good!

File: build.gradle (hunk 1 above)
Comment (jimklimov) [reply build.gradle]: #27

**Comment by jimklimov** [comment]
And just for context, if someone would need a version of this plugin that works for older Jenkins Cores (1.x, 2.x before 2.264 if they can't upgrade their server) it may be possible to revive the adaptive markup from PR #25 which uses tabs or divs depending on core ability. PRs welcome then :)

**Comment by garysturgeon** [comment]
Thanks for doing this! Already downloaded and it is working again.

**Comment by enachi** [comment]
Wow! Finally! Can't wait to try it tomorrow! Thank a lot, everyone!

**Comment by nneul** [comment]
Agreed, thank you very much @yasokada-lc  for putting this fix together and @jimklimov  for getting it merged quickly!

**Comment by jimklimov** [comment]
Thanks @yasokada-lc - fixed the few legacy jobs on our server too, which we had to not touch (just in case to not corrupt the config) since our Jenkins core upgrade ;)

**Pull request merged**